freqtrade/data/dataprovider.py

```diff
@@ -90,8 +90,12 @@ def historic_ohlcv(
         if saved_pair not in self.__cached_pairs_backtesting:
             timerange = TimeRange.parse_timerange(None if self._config.get(
                 'timerange') is None else str(self._config.get('timerange')))
+
+            # In backtesting, the training data was already add
+            add_train_candles = self.runmode in (RunMode.DRY_RUN, RunMode.LIVE)
```

Comment:
that's not a realistic check (it'll always be false).

`historic_ohlcv()` is only called in one place (from `get_pair_dataframe()`), which does not call this method if we're in dry/live mode.